### PR TITLE
Feature/accounts

### PR DESCRIPTION
**Pull request by ansnunez**
- Added storage root and code hash to Account class
- Minor update to TxBlock
- Quickly tested on test_node_simple.h and Test_Account

### CODE REVIEW

File: src/common/Constants.h [hunk 1]
```diff
@@ -43,8 +43,8 @@ const unsigned int CHALLENGE_SIZE = 32;
 const unsigned int RESPONSE_SIZE = 32;
 
 // Acount related sizes
-const unsigned int ACCOUNT_SIZE
-    = UINT256_SIZE + UINT256_SIZE /* + ACC_ADDR_SIZE + PUB_KEY_SIZE*/;
+const unsigned int ACCOUNT_SIZE = UINT256_SIZE + UINT256_SIZE + UINT256_SIZE
+    + UINT256_SIZE /* + ACC_ADDR_SIZE + PUB_KEY_SIZE*/;
```
Comment (kaikawaliu) [inline src/common/Constants.h]: Can you also append what the changes of this size is for into the comment?

File: src/libData/AccountData/Account.cpp [hunk 2]
```diff
@@ -21,9 +21,12 @@ Account::Account(const vector<unsigned char>& src, unsigned int offset)
     }
 }
 
-Account::Account(const uint256_t& balance, const uint256_t& nonce)
+Account::Account(const uint256_t& balance, const uint256_t& nonce,
+                 const uint256_t& storageRoot, const uint256_t& codeHash)
```
Comment (kaikawaliu) [inline src/libData/AccountData/Account.cpp]: Why not use type h256 for the hash data?

**Comment by bb111189** [comment]
Request @Gnnng  review also

File: src/libNode/Node.cpp [hunk 3]
```diff
@@ -53,11 +53,14 @@ void addBalanceToGenesisAccount()
...
     const uint256_t nonce{0};
+    const uint256_t storageRoot{0};
+    const uint256_t codeHash{0};
 
     for (auto& walletHexStr : GENESIS_WALLETS)
     {
         Address addr{DataConversion::HexStrToUint8Vec(walletHexStr)};
-        AccountStore::GetInstance().AddAccount(addr, bal, nonce);
+        AccountStore::GetInstance().AddAccount(
+            addr, Account(bal, nonce, storageRoot, codeHash));
```
[the remark this answers is not in the document]
Comment (kaikawaliu) [reply src/libNode/Node.cpp]: Didn't notice it, will change that

**Comment by kaikawaliu** [comment]
Please review the changes @bb111189 @Gnnng

File: src/libDirectoryService/FinalBlockPreProcessing.cpp [hunk 4]
```diff
@@ -177,7 +177,7 @@ void DirectoryService::ComposeFinalBlockCore()
                       blockNum, timestamp, microblockTrieRoot, stateRoot,
                       numTxs, numMicroBlocks, m_mediator.m_selfKey.second,
                       lastDSBlockNum, dsBlockHeader, m_viewChangeCounter),
-        isMicroBlockEmpty, microBlockTxHashes,
+        vector<bool>(isMicroBlockEmpty), vector<TxnHash>(microBlockTxHashes),
```
Comment (Gnnng) [inline src/libDirectoryService/FinalBlockPreProcessing.cpp]: Just found another place that could use {} initialization.

Comment (kaikawaliu) [reply src/libDirectoryService/FinalBlockPreProcessing.cpp]: Will update in a later pr

**Pull request merged**